### PR TITLE
Change delegate macro to be `delegate *methods, to: object`

### PR DESCRIPTION
**Pull request by asterite**
This is the way I wanted `delegate` to work in the first place, though the language couldn't express that.

This is a breaking change, but it makes code easier to read and understand:

```crystal
delegate foo, bar, baz
# vs.
delegate foo, bar, to: baz
```

Additionally, an extra overload that receives a block is added, which forwards all block arguments, so this covers now all cases except that when a block is captured.

### CODE REVIEW

File: src/string.cr [hunk 1]
```diff
@@ -730,7 +730,7 @@ class String
   end
 
   def at(index : Int)
-    at(index) { raise IndexError }
+    at(index) { raise IndexError.new }
```
Comment (bcardiff) [inline src/string.cr]: Why this change is here?

Comment (asterite) [inline src/string.cr]: Ah, it was wrong. Maybe I should commit the fix to master instead :-P
